src/components/ButtonGroup.tsx

```diff
@@ -0,0 +1,45 @@
+import clsx from 'clsx';
+import { ReactNode } from 'react';
+import { RadioGroup } from '@headlessui/react';
+
+interface ButtonGroupProps {
+  children: ReactNode;
+  value: any;
+  onChange: (selected: any) => void;
+}
+
+export function ButtonGroup({ children, value, onChange }: ButtonGroupProps): JSX.Element {
+  return (
+    <RadioGroup
+      value={value}
+      onChange={onChange}
+      className="flex flex-row items-center justify-start gap-2 rounded-full border border-gray-800 px-1 py-1"
+    >
+      {children}
+    </RadioGroup>
+  );
+}
+
+interface ButtonGroupButtonProps {
+  children: ReactNode;
+  active?: boolean;
+  value: ReactNode;
+}
```

Comment:
Combine with suggestion about generic type params above (and dont use `ReactNode` as the type)

```suggestion
interface ButtonGroupButtonProps<T> {
  children: ReactNode;
  value: T;
}
```

(also suggesting to remove `active` since you dont appear to be using it)

Reply:
applied